app/src/main/java/com/randomappsinc/studentpicker/models/ListInfo.java

```diff
@@ -90,6 +90,18 @@ public void removeNames(String name, int amount) {
         }
     }
 
+    public void setNameAmount(String name, int newAmount) {
+        if (nameAmounts.containsKey(name)) {
+            if (newAmount != 0) {
+                nameAmounts.put(name, newAmount);
+            } else {
+                nameAmounts.remove(name);
+                names.remove(name);
+            }
+            numInstances = newAmount;
```

Comment:
numInstances is the total number of names in the list, so I think this change will break the number of names indicator at the top. We need to change it by the delta we incurred on the name (either add or remove). We will need to get the current number of instances of this name from the HashMap to compute this.